libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -39,6 +42,23 @@ export class BookSearchComponent {
 
   addBookToReadingList(book: Book) {
     this.store.dispatch(addToReadingList({ book }));
+
+    this.store.select(confirmedAddToReadingList)
+      .pipe(
+        take(1)
+      )
+      .subscribe(b => {
+        const mySnackBar = this._snackBar.open(`${book.title} added to reading list`, 'Undo');
+        mySnackBar.afterDismissed().subscribe(info => {
+          if (info.dismissedByAction === true) {
+            this.store.dispatch(removeFromReadingList({
+              item: {
+                bookId: book.id, ...book
+              }
+            }));
+          }
+        });
+      });
```

Comment:
As you noted yourself, this is not a very good approach. Every time a user adds a book to their reading list, a new subscription to the store would be created. I know that you're using the `.take(1)` operator, which would potentially save you from a memory leak, and stop the subscription, still I think that a different approach would be better.

If you wanted to use the `this.store.select...` subscription method, I would place the subscription outside the `addBookToReadingList` function. Essentially, I want to show a snackbar to undo an action, every time `confirmedAddToReadingList` happens. Maybe, then, an even better approach would be to add this "watcher" inside the effects, where you are already using the actions!